src/runtime_src/xdp/profile/plugin/aie_profile/aie_plugin.cpp

```diff
@@ -991,118 +993,450 @@ namespace xdp {
+          } else if (XAIE_PL_MOD == mod) {
+            tiles = getAllTilesForShimProfiling(handle, metrics[i][1]);
+            allGraphsDone = true; // ??
+          }
+        } // allGraphsDone
+        for (auto &e : tiles) {
+          mConfigMetrics[moduleIdx][e] = metrics[i][1];
+          // check if same tiles are recognized
+        }
+      }
+    } // Pass 1 
+
+    // Pass 2 : process only range of tiles metric setting 
+    for (size_t i = 0; i < metricsSettings.size(); ++i) {
+
+      if (XAIE_CORE_MOD == mod || XAIE_MEM_MOD == mod) {
+        if (3 != metrics[i].size()) {
+          continue;
+        }
+        for (size_t j = 0; j < metrics[i].size(); ++j) {
+          boost::replace_all(metrics[i][j], "{", "");
+          boost::replace_all(metrics[i][j], "}", "");
+        }
+        std::vector<std::string> minTile;
+        boost::split(minTile, metrics[i][0], boost::is_any_of(","));
+        uint32_t minCol = std::stoi(minTile[0]);
+        uint32_t minRow = std::stoi(minTile[1]);
+
+        std::vector<std::string> maxTile;
+        boost::split(maxTile, metrics[i][1], boost::is_any_of(","));
+        uint32_t maxCol = std::stoi(maxTile[0]);
+        uint32_t maxRow = std::stoi(maxTile[1]);
+  
+        for (uint32_t col = minCol; col <= maxCol; ++col) {
+          for (uint32_t row = minRow; row <= maxRow; ++row) {
+            xrt_core::edge::aie::tile_type tile;
+            tile.col = col;
+            tile.row = row;
+//            tiles.push_back(tile);
+            mConfigMetrics[moduleIdx][tile] = metrics[i][2];
+            // check if same tiles are recognized
+          }
+        }
+      } else if (XAIE_PL_MOD == mod) {
+        if (3 != metrics[i].size()) {
+          continue;
+        }
+        // Implement
+      } 
+      /* Note : mem_tile_metrics is not handled now. Add handling later
+       */
+    } // Pass 2 
+
+    // Pass 3 : process only single tile metric setting 
+    for (size_t i = 0; i < metricsSettings.size(); ++i) {
+
+      if (XAIE_CORE_MOD == mod || XAIE_MEM_MOD == mod) {
+        if (2 != metrics[i].size()) {
+          continue;
+        }
+        if (0 == metrics[i][0].compare("all")) {
+          continue;
+        }
+        boost::replace_all(metrics[i][0], "{", "");
+        boost::replace_all(metrics[i][0], "}", "");
+
+        std::vector<std::string> tilePos;
+        boost::split(tilePos, metrics[i][0], boost::is_any_of(","));
+
+        xrt_core::edge::aie::tile_type tile;
+        tile.col = std::stoi(tilePos[0]);
+        tile.row = std::stoi(tilePos[1]);
+//        tiles.push_back(tile);
+        mConfigMetrics[moduleIdx][tile] = metrics[i][1];
+      } else if (XAIE_PL_MOD == mod) {
+        if (2 != metrics[i].size()) {
+          continue;
+        }
+        if (0 == metrics[i][0].compare("all")) {
+          continue;
+        }
+        // Implement
+      } 
+      /* Note : mem_tile_metrics is not handled now. Add handling later
+       */
+    } // Pass 3 
```

Comment:
Similar to trace, we should be able to simplify this code using nested loops instead of a series of sequential loops.